lib/worker.js

```diff
@@ -84,30 +80,78 @@ async function getLocalLinter(basedir) {
             linter = require('loophole').allowUnsafeNewFunction(() => require(linterPath).Linter);
           }
         } else {
-          linter = tslintDef;
+          return resolve();
         }
-        tslintCache.set(basedir, linter);
+        tslintCache.set(fileDir, linter);
         return resolve(linter);
       },
     );
   });
 }
 
+function getNodePrefixPath() {
+  return new Promise((fulfill, reject) => {
+    const npmCommand = process.platform === 'win32' ? 'npm.cmd' : 'npm';
+    ChildProcess.exec(
+      `${npmCommand} get prefix`,
+      { env: Object.assign(Object.assign({}, process.env), { PATH: getPath() }) },
+      (err, stdout, stderr) => {
+        if (err || stderr) {
+          reject(err || new Error(stderr));
+        } else {
+          fulfill(stdout.trim());
+        }
+      },
+    );
+  });
+}
+
 async function getLinter(filePath) {
   const basedir = path.dirname(filePath);
   if (tslintCache.has(basedir)) {
     return tslintCache.get(basedir);
   }
 
-  // Initialize the default instance if it hasn't already been initialized
-  loadDefaultTSLint();
-
   if (config.useLocalTslint) {
-    return getLocalLinter(basedir);
+    const localLint = await resolveAndCacheLinter(basedir);
+    if (localLint) {
+      return localLint;
+    }
+  }
+
+  if (fallbackLinter) {
+    tslintCache.set(basedir, fallbackLinter);
+    return fallbackLinter;
+  }
+
+  if (config.useGlobalTslint) {
+    if (config.globalNodePath) {
+      const globalLint = await resolveAndCacheLinter(basedir, config.globalNodePath);
+      if (globalLint) {
+        fallbackLinter = globalLint;
+        return globalLint;
+      }
+    }
+
+    const globalLint = await getNodePrefixPath()
+      .then(prefix => resolveAndCacheLinter(basedir, prefix))
```

Comment:
:bug: Why is this using `Promise` syntax? Nothing in here is a callback function that requires a `Promise` wrapper to work...

Reply:
I guess the `.then()` isn't 100% necessary, I just figured it was easier to chain the two promises and `await` them together, but since `resolveAndCacheLinter()` never rejects, I guess it doesn't really have to be before the `catch()`. It could also be done like:

```js
const prefix = await getNodePrefixPath().catch((err) => {
...
});
if (prefix) {
    const globalLint = await resolveAndCacheLinter(basedir, prefix);
    fallbackLinter = globalLint;
    return globalLint;
}
```

I guess it's just preference as to which makes the flow more obvious? I'd be okay with either.